packages/go_router/lib/src/route.dart

```diff
@@ -443,6 +453,10 @@ class ShellRoute extends RouteBase {
         assert(route.parentNavigatorKey == null);
       }
     }
+    if (nestedNavigationBuilder != null) {
```

Comment:
This and the above for loop should all be in a wrapper assert

Reply:
👍 Fixed in PartitionedShellRoute in upcoming commit.